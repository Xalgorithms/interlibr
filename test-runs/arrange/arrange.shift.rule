META
  VERSION "0.0.1"
  MAINTAINER "Don Kelly <karfai@gmail.com>";

REQUIRE refine:table0:0.0.1 AS table0;

ARRANGE table0 AS table_shift_r3 USING shift(3);
ARRANGE table0 AS table_shift_l9 USING shift(-9);
ARRANGE table0 AS table_invert USING invert();
