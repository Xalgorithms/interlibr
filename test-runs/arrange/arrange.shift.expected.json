{
  "tables" : {
    "table_shift_r3" : [
      { "a" : 5, "b" : "F", "c": "A" },
      { "a" : 4, "b" : "A", "c": "Q" },
      { "a" : 2, "b" : "A", "c": "P" },
      { "a" : 1, "b" : "C", "c": "A" },
      { "a" : 3, "b" : "E", "c": "B" }
    ],
    "table_shift_l9" : [
      { "a" : 2, "b" : "A", "c": "P" },
      { "a" : 1, "b" : "C", "c": "A" },
      { "a" : 3, "b" : "E", "c": "B" },
      { "a" : 5, "b" : "F", "c": "A" },
      { "a" : 4, "b" : "A", "c": "Q" }
    ],
    "table_invert" : [
      { "a" : 2, "b" : "A", "c": "P" },
      { "a" : 4, "b" : "A", "c": "Q" },
      { "a" : 5, "b" : "F", "c": "A" },
      { "a" : 3, "b" : "E", "c": "B" },
      { "a" : 1, "b" : "C", "c": "A" }
    ]
  }
}
